apps/studio/electron/main/code/diff/helpers.ts

```diff
@@ -79,3 +81,27 @@ export const jsxFilter = (
 export function generateCode(ast: t.File, options: GeneratorOptions, codeBlock: string): string {
     return generate(ast, options, codeBlock).code;
 }
+
+export function getDynamicType(elementPath: NodePath<t.JSXElement>): DynamicType | null {
```

Comment:
My understanding is that this traverse up the entire element tree, is this too much? I would think if the immediate parent is an array then we will add it, otherwise we don't. In this case, we don't want to delete a but do we delete b and c?

Perhaps once we run into another parent that is a jsxelement, we can return.

```jsx
<>{arr.map(_ =>
   <a>
     <b/>
     <c/>
   <a/>
)}</>
```

Reply:
``` javascript

export function getDynamicType(elementPath: NodePath<t.JSXElement>): DynamicType | null {
    const parentPath = elementPath.parentPath;
    
    if (t.isJSXElement(parentPath.node)) {
        return null; // Stop if we hit another JSX element
    }


    if (parentPath && t.isCallExpression(parentPath.node) &&
        t.isMemberExpression(parentPath.node.callee) &&
        t.isIdentifier(parentPath.node.callee.property) &&
        parentPath.node.callee.property.name === 'map') {
        return 'array';
    }

    return null;
}
```

@Kitenite does this solve the concern with returning early if the parent is another jsx element?

Reply:
Never mind, it didn't work as expected. root element in the array were not flagged as dynamic